### PR TITLE
Fix for declaring test case in data provider

### PR DESCRIPTION
**Pull request by kubawerlos**
https://github.com/sebastianbergmann/phpunit/issues/2833

### CODE REVIEW

**Comment by kubawerlos** [comment]
Ping @sebastianbergmann

**Comment by sebastianbergmann** [comment]
Can you please send a pull request against `5.7`? Thanks!

**Comment by kubawerlos** [comment]
@sebastianbergmann done

File: src/Framework/TestSuite.php [hunk 1]
```diff
@@ -292,6 +292,12 @@ public function addTestSuite($testClass)
      */
     public function addTestFile($filename)
     {
+        static $declaredClasses;
```
Comment (sebastianbergmann) [inline src/Framework/TestSuite.php]: Please do not use a method-local, static variable here.

Comment (kubawerlos) [reply src/Framework/TestSuite.php]: @sebastianbergmann done, changed to private, dynamic one.

**Pull request merged**